### PR TITLE
Modernizing Team

### PR DESCRIPTION
**Pull request by SJuliez**
This is a refactor of Team which was full of outdated code. Changes in other classes are due to renames or removing Enumerations.

There are two 
// TODO: remove from Team 
in there. I'll remove this in other PRs that are already part-finished.

### CODE REVIEW

File: megamek/src/megamek/client/ui/swing/lobby/TeamOverviewPanel.java [hunk 1]
```diff
@@ -202,7 +202,7 @@ public void updateTable(Game game) {
                 int hiddenBv = 0;
                 boolean[] unitCritical = { false, false, false, false, false };
                 boolean[] unitWarnings = { false, false, false, false, false };
-                for (Player teamMember: team.getPlayersVector()) {
+                for (Player teamMember: team.players()) {
```
Comment (Windchild292) [inline megamek/src/megamek/client/ui/swing/lobby/TeamOverviewPanel.java:205]:
```suggestion
                for (Player teamMember : team.players()) {

```

File: megamek/src/megamek/common/Team.java [hunk 2]
```diff
@@ -1,116 +1,104 @@
...
 
-    public void addPlayer(Player p) {
-        players.addElement(p);
-    }
-    
-    public boolean isObserverTeam() {
-        if (ObserverTeam == null) {
-            cacheObserverStatus();
+    /** Adds the given player to this team. */
+    public void addPlayer(Player player) {
```
Comment (Windchild292) [inline megamek/src/megamek/common/Team.java:70]: nullable annotation

**Review by Windchild292 — COMMENTED** [review]
Two minor changes requested

**Comment by SJuliez** [comment]
Done.

**Review by Windchild292 — APPROVED** [review]

**Pull request merged**